examples/xds-management-server/pkg/resources/resources.go

```diff
@@ -0,0 +1,199 @@
+package resources
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"github.com/golang/protobuf/ptypes"
+	"google.golang.org/protobuf/types/known/wrapperspb"
+	"strconv"
+
+	envoycluster "github.com/envoyproxy/go-control-plane/envoy/config/cluster/v3"
+	envoycore "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
+	envoyendpoint "github.com/envoyproxy/go-control-plane/envoy/config/endpoint/v3"
+	envoylistener "github.com/envoyproxy/go-control-plane/envoy/config/listener/v3"
+	"github.com/envoyproxy/go-control-plane/pkg/cache/types"
+	"github.com/envoyproxy/go-control-plane/pkg/cache/v3"
+	"github.com/golang/protobuf/jsonpb"
+	structpb "github.com/golang/protobuf/ptypes/struct"
+
+	quilkinfilter "quilkin.dev/xds-management-server/filters/debug/v1alpha1"
+)
+
+// Endpoint represents an xds endpoint
+type Endpoint struct {
```

Comment:
Coming from comments in https://github.com/googleforgames/quilkin/issues/346#issuecomment-898384422 - I'm trying to work out what Endpoint, Cluster, FilterConfig, Resource etc are for.

I _think_ it's so that we can specify in a type what our subset of supported xDS is, and thereby not have to deal with all of xDS types in our system?

And then the `make*` commands do all the work to convert our xDS surface into the bigger, more complicated xDS types.

Is that correct?

Reply:
Yes that was the idea, so that we only have one place to do the mapping to xds types.